src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -5263,14 +5263,13 @@ private static Class<?>[] insertArgumentsChecks(MethodHandle target, int insCoun
      *                  or if the new method handle's type would have too many parameters
      */
     public static MethodHandle dropArguments(MethodHandle target, int pos, List<Class<?>> valueTypes) {
-        return dropArguments0(target, pos, copyTypes(valueTypes.toArray()));
+        return dropArguments(target, pos, valueTypes.toArray(new Class<?>[0]), false);
     }
 
-    private static List<Class<?>> copyTypes(Object[] array) {
-        return Arrays.asList(Arrays.copyOf(array, array.length, Class[].class));
-    }
-
-    private static MethodHandle dropArguments0(MethodHandle target, int pos, List<Class<?>> valueTypes) {
+    static MethodHandle dropArguments(MethodHandle target, int pos, Class<?>[] valueTypes, boolean trusted) {
```

Comment:
Having a boolean that flips the behaviour makes it harder to know what that `true` or `false` literal at the call site does. I'd suggest splitting this into `dropArgumentsTrusted`, which doesn't clone the array, and `dropArguments` which always makes a copy (and the latter calls former). WDYT?

Reply:
Yeah, that sounds better.